test/iterators.jl

```diff
@@ -841,3 +868,19 @@ end
     @test cumprod(x + 1 for x in 1:3) == [2, 6, 24]
     @test accumulate(+, (x^2 for x in 1:3); init=100) == [101, 105, 114]
 end
+
+@testset "Iterators.tail_if_any" begin
+    @test Iterators.tail_if_any(()) == ()
+    @test Iterators.tail_if_any((1, 2)) == (2, )
+    @test Iterators.tail_if_any((1, )) == ()
```

Comment:
`grep`ing the Julia files here, it looks like it's much more common to avoid the space after `,`

```suggestion
    @test Iterators.tail_if_any((1, 2)) == (2,)
    @test Iterators.tail_if_any((1,)) == ()
```